File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] sol = new int[2];
        
        for(int i = 1; i <= Math.sqrt(yellow); i++) {
            if(yellow % i == 0) {
                if((i * 2) + ((yellow / i) * 2) + 4 == brown) {
                    sol[0] = (yellow / i) + 2;
                    sol[1] = i + 2;
                    return sol;
                }
            }
        }
        
        return sol;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    width_plus_height = (brown // 2) - 2
    
    for i in range(1, (width_plus_height // 2) + 1):
        h = i
        w = width_plus_height - h
        if h * w == yellow:
            return (w + 2, h + 2)
    
    
# y가로*2 + y세로*2 + 4 = b
# 2(가로 + 세로 + 2) = b